### ISSUE TITLE
dir="auto" does not affect the table tag

### ISSUE THREAD

**Issue by syedkaleem29** [danfickle/openhtmltopdf#123]
It does not cause the table column itself to be rendered from Right to Left. Because of which the layout in the browser and the PDF differs a lot.

Consider a cell which appears on the left side for English. Now when the pdf language changes to Arabic the cell must come on the right and the text within that should be RTL.

But what actually happens is the cell stays left and the text within the cell is RTL.  

i.e., The columns do not run from right to left in tables when the tag is dir="auto"/"rtl"

HTML
![image](https://user-images.githubusercontent.com/11254787/30107658-120106ca-931d-11e7-9b87-b33caf89df11.png)
PDF
![image](https://user-images.githubusercontent.com/11254787/30107703-28c3110a-931d-11e7-9163-484662258af3.png)

**Comment by danfickle** [comment]
Hi @syedkaleem29 

This is noted somewhere, probably in the RTL issue. The problem is that with row spans and col spans, the table code is quite complex.

**Comment by ieugen** [comment]
Hi, any news on this issue?

**Comment by AhmedEzzat12** [comment]
Hello any news? @danfickle

**Comment by danfickle** [comment]
It seems we need to push this up the priority list.

**Comment by AhmedEzzat12** [comment]
> It seems we need to push this up the priority list.

Yes thanks, it's a key feature without it RTL is not really complete it's not only about text.

**Comment by daniil-tagan** [comment]
Hello @danfickle any updates here?

**Comment by wahab321** [comment]
@danfickle  any update about RTL Direction of table?? i need Urgent Can you Please solved in these days

**Comment by mike-mike** [comment]
Hello @wahab321 . If you are looking for urgent soution you may replace the following code in TableSectionBox.java
Inside setCellWidths function  at line 217 write:

**sourse:**
cell.setX(columnPos[j] + hspacing);

**new code for RTL tables only:**
Box parent = getTable().getParent();

if(parent != null){
    int parentWidth = parent.getContentWidth();
    cell.setX(parentWidth - w - (columnPos[j] + hspacing));
}

It solved for me an alignment issue in RTL tables.
I don't know if it is a right solution, but I checked it with simple tables without colspan/rowspan and it works.
At least you can use it until you get a real solution from @danfickle

**Comment by wahab321** [comment]
thanks it work's for me
